OrbitClientGgp/ClientGgp.cpp

```diff
@@ -164,6 +180,43 @@ bool ClientGgp::InitCapture() {
   return true;
 }
 
+void ClientGgp::InformUsedSelectedCaptureFunctions() {
+  if (capture_functions_used_.size() != options_.capture_functions.size()) {
+    for (std::string selected_function : options_.capture_functions) {
+      if (!capture_functions_used_.contains(selected_function)) {
+        ERROR("Function matching %s not found; will not be hooked in the capture",
+              selected_function);
+      }
+    }
+  } else {
+    LOG("All functions provided had at least a match");
+  }
+}
+
+bool ClientGgp::IsSelectedFunction(const FunctionInfo& func) {
+  for (std::string selected_function : options_.capture_functions) {
+    if (func.pretty_name().find(selected_function) != std::string::npos) {
+      if (!capture_functions_used_.contains(selected_function)) {
+        capture_functions_used_.insert(selected_function);
+      }
+      return true;
+    }
+  }
+  return false;
+}
+
+absl::flat_hash_map<uint64_t, FunctionInfo> ClientGgp::GetSelectedFunctions() {
+  absl::flat_hash_map<uint64_t, FunctionInfo> selected_functions;
+  for (const auto& func : target_process_->GetFunctions()) {
+    if (IsSelectedFunction(*func)) {
```

Comment:
I have a question here: In `App.cpp` when selected_functions is populated in _OrbitApp::GetSelectedFunctionsAndOrbitFunctions_, it also adds a function if `FunctionUtils::IsOrbitFunc(*func)` returns true. Reading the code I understand that this check is related only to the Orbit app so the ggp client does not need it but I just want to be sure. Can you please confirm? Thanks!

Reply:
I think this has something to do with the manual instrumentation Pierric is working on. The func.type is KNone for all dynamically instrumented functions I think. The proto would profit from some comments there :)
At some point we'll want the signals from manual instrumentation as well but think it is totally sensible to do this later.

Reply:
I see, thanks for the clarification. If we add the signals from manual instrumentation I guess it will also require some extra work in the client anyways so I think it will make sense to do this change then. Will leave it as it is for now then, thanks!